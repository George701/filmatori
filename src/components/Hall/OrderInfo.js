import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { cleanOrderList } from '../../actions/order';
import { occupyPlaces } from '../../actions/cinema';

export const OrderInfo = ({ cleanOrderList, occupyPlaces, order: { totalPrice, orderPlaces } }) => {

    const onClick = () => {
        parseFloat(totalPrice) !== 0 &&
            occupyPlaces() && cleanOrderList()
    }

    return (
        <div className='order-block'>
            <div className="order-unit">
                <span className="order-unit-title">
                    Seats:
            </span>
                <span className="order-places order-info-num">
                    {` ${orderPlaces.length}`}
                </span>
            </div>
            <div className="order-unit">
                <span className="order-unit-title">
                    Price:
                </span>
                <span className="order-price order-info-num">
                    {` ${totalPrice}$`}
                </span>
            </div>
            <div className="order-unit">
                <button className={parseFloat(totalPrice) === 0 ? 'btn-disabled' : 'order-btn-buy'} onClick={() => onClick()}>
                    Buy
                </button>
            </div>
        </div>
    )
}

OrderInfo.propTypes = {
    order: PropTypes.object.isRequired,
    cleanOrderList: PropTypes.func.isRequired,
    occupyPlaces: PropTypes.func.isRequired
}

const mapStateToProps = state => ({
    order: state.order
});

export default connect(mapStateToProps, { cleanOrderList, occupyPlaces })(OrderInfo)
