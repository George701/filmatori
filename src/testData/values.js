export const propsArrived = {
    cinema: {
        hall: {
            hallName: 'Test Hall',
            city: "Test city",
            date: "2019-11-11T11:11:11.942+00:00",
            movie: "Test movie",
            rows: [
                {
                    _id: "RT01",
                    rowNumber: 1,
                    skip: false,
                    places: []
                },
                {
                    _id: "RT02",
                    rowNumber: 2,
                    skip: true,
                    places: []
                },
            ]
        },
        loading: false
    }
};

export const propsPending = { cinema: { hall: { hallName: '', rows: [] }, loading: true } };

export const orderInfoMiscPositive = {
    totalPrice: 15.45,
    orderPlaces: [
        {
            rowId: "RT01",
            placeId: "PT01",
            rowNumber: 1,
            placeNumber: 1,
        }
    ]
};

export const orderInfoMiscNegative = {
    totalPrice: 0,
    orderPlaces: []
};

export const rowProps = {
    row: {
        _id: 'RT01',
        rowNumber: 1,
        skip: true,
        places: [
            {
                "_id": "PT09",
            },
            {
                "_id": "PT10",
            },
        ]
    }
};

export const placeProps = {
    placeId: 'PT01',
    rowId: "RT01",
    rowNumber: 1,
    place: {
        _id: "PT01",
        placeNumber: 1,
        isReserved: false,
        isOccupied: false,
        skipLeft: 1,
        skipRight: 1,
        type: "chair",
        price: "10$"
    }
};

export const orderPayload = {
    rowNumber: 1,
    placeNumber: 1,
    r_id: 'RT01',
    p_id: 'PT01',
    price: 15.50
};

export const cinemaInitialState = {
    loading: true,
    hall: {
        hallName: '',
        city: '',
        movie: '',
        date: '',
        rows: []
    }
};

export const cinema = {
    loading: false,
    hall: {
        hallName: 'Test Hall',
        city: '',
        movie: '',
        date: '',
        rows: [
            {
                _id: "RT01",
                rowNumber: 1,
                places: [
                    {
                        _id: "PT01",
                        placeNumber: 1,
                        isReserved: false,
                        isOccupied: false
                    }
                ]
            }
        ]
    }
};

export const orderInitialState = {
    orderPlaces: [],
    totalPrice: 0
};

export const orderReducerPayload = {
    rowNumber: 1,
    placeNumber: 1,
    rowId: 'RT01',
    placeId: 'PT01',
    price: 15.50
};