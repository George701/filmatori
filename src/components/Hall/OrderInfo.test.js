import React from 'react';
import { shallow } from '../../enzyme';
import { OrderInfo } from './OrderInfo';
import { orderInfoMiscPositive, orderInfoMiscNegative } from '../../testData/values';

describe('OrderInfo', () => {
    const cleanOrderList = jest.fn();
    const occupyPlaces = jest.fn();

    const propsPositive = {
        cleanOrderList,
        occupyPlaces,
        order: orderInfoMiscPositive
    }

    const orderInfo = shallow(<OrderInfo {...propsPositive} />);

    it('renders properly', () => {
        expect(orderInfo).toMatchSnapshot();
    });

    it('displays correct amount of seats', () => {
        expect(parseInt(orderInfo.find('.order-places').text(), 10)).toEqual(propsPositive.order.orderPlaces.length);
    });

    it('displays correct price', () => {
        expect(orderInfo.find('.order-price').text()).toEqual(` ${propsPositive.order.totalPrice}$`);
    });

    it('displays active button', () => {
        expect(orderInfo.find('.order-btn-buy').exists()).toBe(true);
    });

    describe('when clicking the button', () => {
        beforeEach(() => {
            orderInfo.find('.order-btn-buy').simulate('click');
        });

        // it('uses a function to record booked seats', () => {
        //     expect(occupyPlaces).toHaveBeenCalledTimes(1);
        // });
    });

    describe('after clicking button', () => {
        const propsNagative = {
            cleanOrderList,
            occupyPlaces,
            order: orderInfoMiscNegative
        }

        const orderInfoAfter = shallow(<OrderInfo {...propsNagative} />);

        it('displays correct amount of seats', () => {
            expect(parseInt(orderInfoAfter.find('.order-places').text(), 10)).toEqual(0);
        });

        it('displays correct price', () => {
            expect(orderInfoAfter.find('.order-price').text()).toEqual(` ${0}$`);
        });

        it('does not display active button', () => {
            expect(orderInfoAfter.find('.order-btn-buy').exists()).toBe(false);
        });

        it('displays disabled button', () => {
            expect(orderInfoAfter.find('.btn-disabled').exists()).toBe(true);
        });
    });
});
