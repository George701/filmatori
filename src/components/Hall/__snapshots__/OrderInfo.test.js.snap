// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderInfo renders properly 1`] = `ShallowWrapper {}`;
